discord/sink.py

```diff
@@ -0,0 +1,775 @@
+import struct
+import wave
+import os
+import threading
+import logging
+import subprocess
+from collections import namedtuple
+from typing import TYPE_CHECKING, Optional, Callable, Any, IO, Union
+
+from .opus import Decoder as OpusDecoder
+from .errors import ClientException
+from .enums import RTCPMessageType
+from .utils import strip_namedtuple_docs
+
+if TYPE_CHECKING:
+    from .member import Member
+    from .voice_client import VoiceClient
+
+
+__all__ = (
+    "AudioFrame",
+    "AudioSink",
+    "AudioFileSink",
+    "WaveAudioFileSink",
+    "MP3AudioFileSink",
+    "RTCPPacket",
+    "RTCPSenderReportPacket",
+    "RTCPReceiverReportPacket",
+    "RTCPSourceDescriptionPacket",
+    "RTCPGoodbyePacket",
+    "RTCPApplicationDefinedPacket",
+    "RTCPReceiverReportBlock",
+    "RTCPSourceDescriptionChunk",
+    "RTCPSourceDescriptionItem"
+)
+
+
+_log = logging.getLogger(__name__)
+
+
+RTCPReceiverReportBlock = namedtuple("RTCPReceiverReportBlock", ("ssrc", "f", "c", "ehsn", "j", "lsr", "dlsr"))
```

Comment:
I believe it would be better to use `typing.NamedTuple` for clearer types?

Reply:
Yeah that's true. I'll switch to using `typing.NamedTuple`.